### PR TITLE
Don't show signed addon warning if we're not signing (bug 1135926)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1135926](https://bugzilla.mozilla.org/show_bug.cgi?id=1135926)

### CODE REVIEW

**Comment by magopian** [comment]
@kmaglione, @clouserw r?

File: apps/devhub/tasks.py [hunk 1]
```diff
@@ -51,6 +54,20 @@ def validator(upload_id, **kw):
...
+    """Remove the "Package already signed" warning if we're not signing."""
+    try:
+        result = json.loads(result_json)
+        messages = result['messages']
+    except (KeyError, ValueError):
+        return result_json
+    messages = [m for m in messages if 'signed_xpi' not in m['id']]
+    if len(messages) != len(result['messages']):  # We did remove a warning.
+        result['messages'] = messages
+        result['warnings'] -= 1
```
Comment (kmaglione) [inline apps/devhub/tasks.py]: Shouldn't this be something ilke `diff = len(result['messages']) - len(messages); if diff: ... result['warnings'] -= diff` ?

Comment (magopian) [inline apps/devhub/tasks.py]: you mean if there's other `signed_xpi` warnings in the future? Good point, will change that, thanks!

**Comment by kmaglione** [comment]
I'd much rather this be a validator flag to skip the tests rather than a filter in Olympia. But I guess since it's only temporary, it's not a big deal.

r+wc

File: apps/devhub/tasks.py (hunk 1 above)
Comment (kmaglione) [inline apps/devhub/tasks.py]: I could be wrong, but I suspect that there may be multiple `signed_xpi` warnings already, for multi-package XPIs.

**Pull request merged**